src/serializer/ResidualHeapSerializer.js

```diff
@@ -324,26 +336,32 @@ export class ResidualHeapSerializer {
   _emitProperty(
     val: ObjectValue,
     key: string | SymbolValue,
-    desc: Descriptor,
+    desc: Descriptor | void,
     cleanupDummyProperty: boolean = false
```

Comment:
Rename this parameter. It no longer just applies to some "dummy" property. A better name now would be `deleteIfMightHaveBeenDeleted`.